### PR TITLE
[RFC] vim-patch:8.0.0142

### PR DESCRIPTION
**Pull request by ckelsel**
Problem:    Normal colors are wrong with 'termguicolors'.
Solution:   Initialize to INVALCOLOR instead of zero. (Ben Jackson, closes
            vim/vim#1344)

https://github.com/vim/vim/commit/0cdb72aa38c4a0140c94d56bf8bc17cb30260ebf

### CODE REVIEW

**Comment by justinmk** [comment]
Can someone test this?

**Comment by KillTheMule** [comment]
Steps are laid out in https://github.com/vim/vim/issues/1343, I'd give it a shot if no one else wants to.

**Comment by teto** [comment]
I've had a bunch of similar error I fixed in my hilights-related PR ; https://github.com/neovim/neovim/pull/7082  and subsequent. It might or not fix this already.

**Comment by KillTheMule** [comment]
I can't reproduce what's described in https://github.com/vim/vim/issues/1343#issuecomment-269716376, neither in vim (I downloaded 8.0.0141 for the test), nor in nvim. Also the corresponding test does not show a change when run on master vs. run with this PR. You can see the test at https://github.com/KillTheMule/neovim/commit/01fcdbd84d54f6af84f8f21392b8153172582758.

Not sure how to proceed when I can't reproduce the issue.

**Comment by jamessan** [comment]
> Ok I could reproduce with vim 8.0.0134, but still not with nvim.

Yes, nvim behaves differently than vim.  It behaves like gvim does in this case.  However, IMHO, both nvim and gvim are wrong.  They show (bold?) white text on a white background.  The background should be black.

**Comment by KillTheMule** [comment]
It's not black, it's grey right? That's the cursorline, if I'm not mistaken.

**Comment by jamessan** [comment]
> It's not black, it's grey right?

Ah, you're right.  nvim and gvim set `guibg` to `Grey90` for `CursorLine`, while vim sets it to `Grey40`.

**Comment by jamessan** [comment]
> nvim and gvim set guibg to Grey90 for CursorLine, while vim sets it to Grey40.

Which is due to differences in the value for `'background'`.  `background=dark` sets `Grey40`, `background=light` sets `Grey90`.  gvim, and apparently nvim, default to light and vim defaults to dark.

File: src/nvim/syntax.c [hunk 1]
```diff
@@ -6827,6 +6827,9 @@ int hl_combine_attr(int char_attr, int prim_attr)
     new_en = *char_aep;
   } else {
     memset(&new_en, 0, sizeof(new_en));
+    new_en.rgb_fg_color = -1;
```
Comment (bfredl) [inline src/nvim/syntax.c]: Alternatively, maybe this entire branch can be removed? `new_en` already has the correct `ATTRENTRY_INIT` value before the `memset()`.

**Review by justinmk — CHANGES_REQUESTED** [review]
see comments

**Comment by ckelsel** [comment]
Fixed.

**Review by justinmk — APPROVED** [review]

**Pull request merged**